In [1]:
import numpy as np

# Тема 6 "Понятие о производной"

### 1. Найти производную выражения:

 - **a.** $sin(x) \cdot cos(x)$
 
$$f(x) = sin(x) \cdot cos(x)$$

$$f'(x) = \left( sin(x) \cdot cos(x) \right )' =\$$

$$= (sin(x))' \cdot cos(x) + sin(x) \cdot (cos(x))' =\$$

$$= cos(x) \cdot cos(x) + sin(x) \cdot (-sin(x)) =\$$

$$= cos^{2}(x) - sin^{2}(x)$$

**Ответ: $f'(x) = cos^{2}(x) - sin^{2}(x) $**

 - **b.** $ln(2x+1)^{3}$
 
$$f(x) = ln(2x+1)^{3} = 3 \cdot ln(2x+1)$$

$$f'(x) = 3 \cdot \left( ln(2x+1)\right )' =\$$

$$= 3 \cdot \frac{1}{2x+1} \cdot 2 =\$$

$$= \frac{6}{2x+1}$$

**Ответ: $f'(x) = \frac{6}{2x+1}$**

 - **c.** $\sqrt{sin^{2}(ln(x^{3}))}$
 
$$f(x) = \sqrt{sin^{2}(ln(x^{3}))} = \left( sin^{2}(3 \cdot ln(x)) \right)^{\frac{1}{2}}$$

$$f'(x) = \left( \left( sin^{2}(3 \cdot ln(x)) \right)^{\frac{1}{2}} \right)' =\$$

$$= \frac{1}{2} \cdot \left( sin^{2}(3 \cdot ln(x)) \right)^{-\frac{1}{2}} \cdot \left( sin^{2}(3 \cdot ln(x)) \right)' =\$$

$$= \frac{1}{2 \cdot \sqrt{sin^{2}(3 \cdot ln(x))}} \cdot 2 \cdot sin(3 \cdot ln(x)) \cdot \left( sin(3 \cdot ln(x)) \right)' =\$$

$$= \frac{sin(3 \cdot ln(x))}{\sqrt{sin^{2}(3 \cdot ln(x))}} \cdot cos(3 \cdot ln(x)) \cdot 3 \cdot \frac{1}{x} =\$$

$$= \frac{sin(3 \cdot ln(x))}{|sin(3 \cdot ln(x))|} \cdot \frac{3 \cdot cos(3 \cdot ln(x))}{x}$$

**Ответ: $f'(x) = \frac{sin(3 \cdot ln(x))}{|sin(3 \cdot ln(x))|} \cdot \frac{3 \cdot cos(3 \cdot ln(x))}{x}$**

 - **d.** $\frac{x^{4}}{ln(x)}$
 
$$f(x) = \frac{x^{4}}{ln(x)}$$

$$f'(x) = \left( \frac{x^{4}}{ln(x)} \right)' = \left( x^{4} \cdot (ln(x))^{-1} \right)' =\$$

$$= (x^{4})' \cdot (ln(x))^{-1} + x^{4} \cdot \left((ln(x))^{-1}\right)' =\$$

$$= 4x^3 \cdot (ln(x))^{-1} + x^{4} \cdot (-1) \cdot \left( ln(x) \right)^{-2} \cdot \frac{1}{x} =\$$

$$= \frac{4x^{3}}{ln(x)} - \frac{x^{3}}{ln^{2}(x)} =\$$

**Ответ: $f'(x) = \frac{4x^{3}}{ln(x)} - \frac{x^{3}}{ln^{2}(x)}$**

### 2. Найти выражение производной функции и ее значение в точке:

$f(x) = cos(x^{2}+3x), x_{0}=\sqrt{\pi}$

$$f'(x) = \left( cos(x^{2}+3x) \right)' = -sin(x^{2}+3x) \cdot (2x+3)$$

$$f'(\sqrt{\pi}) =  -sin \left( \sqrt{\pi^{2}} + 3\sqrt{\pi} \right) \cdot \left(2 \sqrt{\pi} + 3 \right)$$

In [3]:
-np.sin(np.pi + 3 * np.sqrt(np.pi)) * (2*np.sqrt(np.pi) + 3)

-5.383302410890619

**Ответ: $f'(\sqrt{\pi}) \approx -5.383$**

### 3. $\ast$ Найти значение производной функции в точке:

$f(x) = \frac{x^{3}-x^{2}-x-1}{1+2x+3x^{2}-4x^{3}}, x_{0}=0$

$$f'(x) = \left( \frac{x^{3}-x^{2}-x-1}{1+2x+3x^{2}-4x^{3}} \right)' =\$$

$$= \left( (x^{3}-x^{2}-x-1) \cdot (1+2x+3x^{2}-4x^{3})^{-1} \right)' =\$$

$$= (x^{3}-x^{2}-x-1)' \cdot (1+2x+3x^{2}-4x^{3})^{-1} + (x^{3}-x^{2}-x-1) \cdot \left( (1+2x+3x^{2}-4x^{3})^{-1} \right)' =\$$

$$= (3x^{2}-2x-1) \cdot (1+2x+3x^{2}-4x^{3})^{-1} + (x^{3}-x^{2}-x-1) \cdot (-1) \cdot (1+2x+3x^{2}-4x^{3})^{-2} \cdot (2+6x-12x^{2})=\$$

$$= \frac{3x^{2}-2x-1}{1+2x+3x^{2}-4x^{3}} - \frac{(x^{3}-x^{2}-x-1)\cdot(2+6x-12x^{2})}{(1+2x+3x^{2}-4x^{3})^{2}}$$

$$f'(0)= \frac{3\cdot0^{2}-2\cdot0-1}{1+2\cdot0+3\cdot0^{2}-4\cdot0^{3}} - \frac{(0^{3}-0^{2}-0-1)\cdot(2+6\cdot0-12\cdot0^{2})}{ \left( 1+2\cdot0+3\cdot0^{2}-4\cdot0^{3} \right)^{2}} =\$$

$$f'(0)= \frac{-1}{1} - \frac{(-1)\cdot 2}{1} = 1\$$

**Ответ: $f'(0)= 1\$**

### 3. Найти угол наклона касательной к графику функции в точке:

$f(x) = \sqrt{3x} \cdot ln(x), x_{0}=1$

$$f(x) = \sqrt{3x} \cdot ln(x) = \sqrt{3} \cdot \sqrt{x} \cdot ln(x)$$

$$f'(x) = \sqrt{3} \cdot \left( \sqrt{x} \cdot ln(x) \right)' =\$$

$$= \sqrt{3} \cdot \left( \frac{1}{2\sqrt{x}} \cdot ln(x) + \sqrt{x} \cdot \frac{1}{x} \right) =\$$

$$= \sqrt{3} \cdot \left( \frac{ln(x)}{2\sqrt{x}} + \frac{1}{\sqrt{x}} \right) =\$$

$$f'(1)= \sqrt{3} \cdot \left( \frac{ln(1)}{2\sqrt{1}} + \frac{1}{\sqrt{1}} \right) = \sqrt{3} \cdot \left(\frac{0}{1} + 1 \right) = \sqrt3$$

$$\theta_{f(x)} = arctg(f'(x))$$

In [23]:
np.arctan(np.sqrt(3))

1.0471975511965979

In [24]:
np.arctan(np.sqrt(3)) * 180 / np.pi

60.00000000000001

$$\theta_{f(x)} = arctg(\sqrt{3}) = 60_{degrees} = 1.047_{rad}$$